### DISCUSSION TITLE
How to skip the line in case of lexy::dsl::delimited parsing error?

### DISCUSSION THREAD

**Discussion by Life4gal** [foonathan/lexy#169] [Q&A] [answered]
![image](https://github.com/foonathan/lexy/assets/52756109/a57916b2-d333-4784-bc49-d72dcf381af5)

![image](https://github.com/foonathan/lexy/assets/52756109/0c34c0b3-9da2-4d77-b4ee-35ba652524e9)

Minimal repro:
[godbolt](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGISQAcpK4AMngMmAByPgBGmMQgAKwAbKQADqgKhE4MHt6%2B/kEZWY4CYRHRLHEJKbaY9qUMQgRMxAR5Pn6BdQ05za0E5VGx8UmpCi1tHQXdEwNDldVjAJS2qF7EyOwc5gDMWDQRANQh2AAaAJoA%2BsohAIIXAOIASgDycpEAIjdvn3JuABUAJKvSJHNLEDBeBw5EwaO57A7hTAnc7XAASdyEVw%2BQLcr0%2B2CunzuALuFix2COXDhCLMu3CyG8WCOJl2bicE2ImFYbOwtL2jOZKLZbnoqgAngB6US0WgxJjIADWUqY6CYaQI8QAdAg0mk%2BQL6UKvCzReLpbL5YqVVzwsBdfrDfDBQwmaaReyLVL0ApaI6Dbt%2BS7jW7haz2XajM6EfC5o5kEdwgQjmgvIJ4hHPkcNGyrC74cw2Ao0oqUb7aFmjhaQCAK3mBfCpVK2Z9W62jsBiKwWK0s%2B3dp9m7SudCUxDUGl4gQJVclZgJUcQNXMJLawRuwwS61GAQrhP0GOcqyAOz5u5HS9HUcOJMMABuYjw6CuRZFZ9pV6/qYEE1XEOvFoEyOJgvCII43yrcwzCYI5H1oZ9wUhKc2kXSCAHcmAUI5uQARy8PBuXQI4EHiTBoIbeEvxME8212c9PyvNAty1VQAPjPBE1A8DII7aCJxQmcjnnCUKPoxsLyY39WPYoDOJAsDUBwrx6H7Kjv0vCtaywBCWEITB0AgLSQCnTAlQgABaYz004jB2BQWRIUrCyjmshhbKwWsYgMBglSOCzGI0oLgpC1y/VrTB8LEK4smABhVkC0LQuM0zzOMyKvGi2L4uWCBEuCyy3I8%2BzmI3Txlm1eJIWIUVwng59X1Ycig2WSjY0ky9mL/NjiEAphgO4pT4K8T1sxrEAsJijd7VFCZ0FrKNgENcSXVotqRw3MckMnadZ2GlFl3GjdDG3blBH3SFDxhART3PL8upk3qOK4xSIKaqCzDMATdrgsQRrEhj1M66T/yeuSXvA4gVNGsLaG0%2Bo8D0rVDPyq8UswMzCvCkAbLQTyHMEJz/Nh2tcbsryfKVBKge/dHMaKvH7IidCEIiZZcqs7Gyfx0qnNalb2vukGer6gbXv2qtxsmxbZoIeaQBloN1rWgWNqh29vtQ27Eoe0HRfkwa3rYD6vuQ3aAYkoWWL156FMh6GqxS0VNZnOcFz5VlLBJkAMqyvA4s9u6kuDkLjMnAgjOxwM3Bdva/ua7BlkD1GQ9T4zmdZ5rAY6n9rZF23DYljtxqtBVlVFe9UGfPk8pp6jEisRJPggOaFum6MzGScxkmEhdSEA%2BWZc77vfu8TAk4CoM4Kr4iaKD0LW5QKqIzcUUjjAHYG43le18XogpuIe0IDTDMassawk9FNeKKbkxEjcBgKLr1Oryv9lPbMETlzvqxPp39%2BRL/zcB/O%2BbZ76P0%2BinF%2Bb9gHQX2t/Bu0EgGjxGsgm%2BYCH4W2fpeGinxSBQJ/qAlucs26Hw7l3TuvcJT93GgwFStBw4Tw9pXRCc8oFfkXlsYgvUYHr03hYbevC96oAPkfE%2BWoz6/wsJfdk19IG33AU/HOL9X6yPftBL%2BrJEF/14YA3h6C76YMgdg6BajYGfXgRBehSDeEb0MQwOxAtgq4P5gxFW2cbzjkukeG6bCaa63zuDO2Sl0IIH0tuLYjtsZYWQHgCgIBvKGCVOtGmtsAkAUNlDVSHZjISL0swIgxBI5wx9qgKgFVw5XAQhMWuyi0bYw3LOYptZo6xz5PgkxoVmxJjiiQFEBAwnYUzkmKgIEGBHCqiQHCTA8BKHQOw0OXNBB0GaSADOyJ2auMtlJPOsl%2BoG3FvHSWq4JS1n4MQTCxB5nshYdc/kqt3G0mHIOAc2ZezhCOFQdM11xmvNbMOEMDIwwehXt6RUjQpSlmIEoAMMZXTujNF6E5Up7yzMyghAAXpgWFStAUmkRWKE5VxVwEBlKgFgaQ6DxGJWxBh3JiA4uDHSIFCLPSEslDS0l3IMhtGJdwkgjKJLJiOO8hgEBL4fhpobcIaQwLHI5SS2saAKVUuIDStIdLqUyrAuKlJOdMmYAUCpFM7YTHjShUoUUpxLg3HuE8N4HxvjKF%2BP8YEoIa7aoIDQoliqQDcpIHuSZxAtmrTbPCDgqw4YcESLwPwHAtCkFQJwVe58vYKHWJsEU9IeCkAIJoCNqwlRJA0PoTgkhY35sTZwXgCgQAlrzfGiNpA4CwCQKuTAyBFJkAoBAVowAFDKEMPUIQCBUDoTjQm5VlKDCNEHREWgI6x0Tt4FO1VKAfLACuJtXypBV30GIJEJq1bd3kunfEV4YFF3jsre25AdxiD9vgKsW9zR8Bxr0YOFB5F4QfrDXcOqT5iKZ1pL%2Br9ApP0AYQkB5EIGFxVn2rSXg/BBAiDEOwKQMhBCKBUOoRtpBdBcH0IYYw1hrD6DwDEWtsA3wgAtKQe8oxt1U1WOHHItaOAWTmq2UwqaLBmATagBj3DnyGqfWsDYWw9BzXCHO4do7r14fQt2NInAeCRs4DG0gy6q0cGwKoDtXajiqACMkCyyRJCpg3UcCATGk4QBTVI/uuBCBTL2FwZYvAG1aHZqQItiQS1RvLaQFgEgNAlu00mjgNa625vzT5qNZgK14ci55uLqwhNZGcJIIAA%3D)

**Comment by foonathan** [comment, marked as answer]
You can debug it using the `lexy::trace` action: https://lexy.foonathan.net/playground/?id=PdKdvq93K&mode=trace As you can see, it parses a property key beginning it with `invalid line`, then sees the newline, which is not control. However, since the delimited ends on an equal sign, it skips it during error recovery and continues parsing until it finds the next equal sign.

So if you don't want that behavior, have the property key end or a newline. It then continues parsing a property, has a missing equal sign, which is now an unrecoverable error, and you get the desired error recovery in the top-level production:  https://godbolt.org/z/8jfW95bsv (change in line 24).

**Comment by Life4gal** [comment]
It does solve problem 1 (skip invalid lines):
> invalid line

But problem 2 is still not solved (skip lines without property keys):
> = invalid line

It still generates a property, even though property_key is empty.
![image](https://github.com/foonathan/lexy/assets/52756109/a142e7c6-76a2-46c6-88e3-7944635baa25)

**Reply by foonathan** [reply]
Yes, but this is just error recovery? lexy tries its best to give you as much information as there is in the input. If you can't handle empty keys, you can filter them out later or not attempt to process input when there have been any errors.
